src/Compatibility/Microsoft.DotNet.ApiCompatibility/Abstractions/ElementContainer.cs

```diff
@@ -0,0 +1,17 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.DotNet.ApiCompatibility.Abstractions
+{
+    public class ElementContainer<T>
```

Comment:
I started with this as a Generic type even though it is now currently used with `IAssemblySymbol` only as this might be useful to use it to store the elements in the mappers and that way we could customize the diagnostic messages using the metadata information for each element in the mapper.

However I don't know if it is too much overhead and instead just have a setting to customize the `left` and `right` names when raising diagnostics.